c++/src/kj/compat/tls.c++

```diff
@@ -405,31 +416,109 @@ private:
 // =======================================================================================
 // Implementations of ConnectionReceiver, NetworkAddress, and Network as wrappers adding TLS.
 
-class TlsConnectionReceiver final: public kj::ConnectionReceiver {
+template <typename T>
+class InputQueue {
+  // InputQueue is an async FIFO queue that has many sources but only one sink.
+  // TODO(now) Should I find a header home for this somewhere?
+
 public:
-  TlsConnectionReceiver(TlsContext& tls, kj::Own<kj::ConnectionReceiver> inner)
-      : tls(tls), inner(kj::mv(inner)) {}
+  void pushRaw(T&& v) {
+    // Push an existing value onto the queue.
 
-  Promise<Own<AsyncIoStream>> accept() override {
-    return inner->accept().then([this](kj::Own<AsyncIoStream> stream) {
-      return kj::evalNow([&] { return tls.wrapServer(kj::mv(stream)); })
-          .catch_([this](kj::Exception&& e) {
-        KJ_LOG(ERROR, "error accepting tls connection", e);
-        return accept();
-      });
+    if (auto localFulfiller = kj::mv(fulfiller)) {
```

Comment:
`fulfiller` should be a `kj::Maybe`, rather than rely on nullable `Own`.

On another note, you may want to check `fulfiller->isWaiting()` to see whether the acceptor canceled the wait, in which case you should discard the fulfiller and proceed as if it wasn't there. It's not super-likely to be important in practice, but it's good to form a habit of writing cancellation-safe code. See also: https://github.com/capnproto/capnproto/blob/master/kjdoc/tour.md#cancellation

Reply:
That makes sense to me!